File: week1/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int getChange(void);
int countCoins(int cents, int coinValue);

int main(void)
{
    int cents = getChange();
    int coinsQuantity = 0;

    int quarter = 25;
    int dime = 10;
    int nickel = 5;
    int pennies = 1;

    // get the lowest amount of coins possible
    while (cents > 0)
		{
        int coinValue = 0;

        // select the biggest coin possible for the cents restant
        if (cents >= quarter)
            coinValue = quarter;
        else if (cents >= dime)
            coinValue = dime;
        else if (cents >= nickel)
            coinValue = nickel;
        else if (cents >= pennies)
            coinValue = pennies;

        int coins = countCoins(cents, coinValue);
        coinsQuantity += coins;
        cents -= coins * coinValue;
    }

    printf("%i\n", coinsQuantity);
}

// get the cents owed from user's input
int getChange(void)
{
    int cents;

    do {
        cents = get_int("Change owed: ");
    } while (cents < 0);

    return cents;
}

// count the max amount of coins of a specific value can be used
int countCoins(int cents, int coinValue)
{
    int coinsQuantity = 0;

    while (cents >= coinValue)
		{
        coinsQuantity++;
        cents -= coinValue;
    }

    return coinsQuantity;
}

File: week6/sentimental-cash/cash.py
from cs50 import get_float

def main():
    cents = getChange() * 100
    coinsQuantity = 0

    quarter = 25
    dime = 10
    nickel = 5
    pennies = 1

    while cents > 0:
        coinValue = 0

        if cents >= quarter:
            coinValue = quarter
        elif cents >= dime:
            coinValue = dime
        elif cents >= nickel:
            coinValue = nickel
        elif cents >= pennies:
            coinValue = pennies

        coins = countCoins(cents, coinValue)
        coinsQuantity += coins
        cents -= coins * coinValue

    print(coinsQuantity)


def countCoins(cents, coinValue):
    coinsQuantity = 0

    while True:
        coinsQuantity += 1
        cents -= coinValue
        if cents < coinValue:
            break

    return coinsQuantity


def getChange():
    dollars = 0

    while True:
        dollars = get_float("Change owed: ")
        if dollars > 0:
            break

    return dollars


main()